web/src/components/CollapsibleList.js

```diff
@@ -0,0 +1,92 @@
+import PropTypes from 'prop-types';
+import React, { Component, Fragment } from 'react';
+
+class ListItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      expanded: props.initialExpanded
+    };
+  }
+
+  toggle = () => {
+    this.setState(prev => ({ expanded: !prev.expanded }));
+  };
+
+  render() {
+    const { children, deleteItem, header } = this.props;
+
+    const childClass = this.state.expanded ? '' : 'd-none';
+
+    return (
+      <Fragment>
+        <div className="mb1 h5 flex justify-between">
+          <button
+            type="button"
+            className={`btn btn-no-focus p1 flex-auto left-align bg-blue-light ${
+              deleteItem ? 'rounded-left' : 'rounded'
+            }`}
+            onClick={this.toggle}
+          >
+            <header className="flex items-center justify-between">
+              {header}
+            </header>
+          </button>
+          {deleteItem && (
+            <button
+              type="button"
+              onClick={() => deleteItem()}
+              className="btn btn-no-focus p1 bg-blue-light rounded-right"
+            >
+              ✗
+            </button>
+          )}
+        </div>
+        <div className={childClass}>{children}</div>
+      </Fragment>
+    );
+  }
+}
+
+ListItem.propTypes = {
+  children: PropTypes.node,
+  deleteItem: PropTypes.func,
+  header: PropTypes.node.isRequired,
+  initialExpanded: PropTypes.bool
+};
+
+ListItem.defaultProps = {
+  children: null,
+  deleteItem: null,
+  initialExpanded: false
+};
+
+const CollapsibleList = ({ content, deleteItem, getKey, header, items }) => (
+  <div>
```

Comment:
I don't know that much about when to use `Fragment`s, but is this a case where that might be handy? This `div` doesn't seem to do much except keep React from complaining about not having a container div.

Reply:
That's a good point.  We should probably come up with some rules for when to use one or the other.  I was thinking the `div` was a nice way to group the things in the DOM, but I don't know if that's actually useful _at all_.  😛  Certainly might be useful if we wanted to apply styles all the way down, I guess?  I think I'd suggest punting on changing anything until #1077 for this as well, but I think it's a good idea to talk about which we use, and when.  Maybe we could write it down.  In documentation somewhere.  🤔